### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by YanaDubov**

### CODE REVIEW

File: src/main/java/com/yana/internship/entity/Address.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "address")
+public class Address {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+    private String country;// TODO: 2019-11-13 to enum
```
Comment (rfisenko-integrity) [inline src/main/java/com/yana/internship/entity/Address.java]: Please fix the TODO

File: src/main/java/com/yana/internship/entity/Order.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
+package com.yana.internship.entity;
+
+import org.hibernate.annotations.CreationTimestamp;
+
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Table(name = "app_order")
```
Comment (rfisenko-integrity) [inline src/main/java/com/yana/internship/entity/Order.java]: Why we need prefix `app`?

File: src/main/java/com/yana/internship/entity/Order.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @CreationTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
```
Comment (rfisenko-integrity) [inline src/main/java/com/yana/internship/entity/Order.java]: Let's use common data format for all fields. Date or Timestamp

File: src/main/java/com/yana/internship/entity/Order.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Temporal(TemporalType.DATE)
+    @Column(name = "check_in_date")
+    private Date checkInDate;
+
+    @Temporal(TemporalType.DATE)
+    @Column(name = "check_out_date")
+    private Date checkOutDate;
+
+    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```
Comment (rfisenko-integrity) [inline src/main/java/com/yana/internship/entity/Order.java]: Should we remove Apartments if the Order will be removed?

File: src/main/java/com/yana/internship/entity/Order.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    @JoinColumn(name = "apartment_id")
+    private Apartment apartment;
+
+    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```
Comment (rfisenko-integrity) [inline src/main/java/com/yana/internship/entity/Order.java]: The same question as for Apartments

File: src/main/java/com/yana/internship/entity/Tariff.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+@Entity
+@Table(name = "tariff")
+public class Tariff {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+    @Column(name = "cost_per_night")
+    private Integer costPerNight;
```
Comment (rfisenko-integrity) [inline src/main/java/com/yana/internship/entity/Tariff.java]: Let's use BigDecimal

File: src/main/java/com/yana/internship/entity/Tariff.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "tariff")
+public class Tariff {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
```
Comment (rfisenko-integrity) [inline src/main/java/com/yana/internship/entity/Tariff.java]: Let's use Long for all ID's as we decided.

File: src/main/resources/db/migration/V1__1_0__my_first_migration.sql [hunk 8]
```diff
@@ -1,5 +1,5 @@
-CREATE TABLE IF NOT EXISTS `test` (
+CREATE TABLE IF NOT EXISTS test (
```
Comment (rfisenko-integrity) [inline src/main/resources/db/migration/V1__1_0__my_first_migration.sql]: Please remove all unnecessary scripts

File: src/main/resources/db/migration/V4__2_0_user_table.sql [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    CONSTRAINT fk_tariff FOREIGN KEY (tariff_id) REFERENCES tariff(id)
+);
+CREATE TABLE IF NOT EXISTS app_order (
+
+    id int NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name varchar(20),
+    apartment_id int NOT NULL,
+    customer_id int NOT NULL,
+    creation_date timestamp NOT NULL,
+    check_in_date timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
Comment (rfisenko-integrity) [inline src/main/resources/db/migration/V4__2_0_user_table.sql]: Are you sure that we should have default values for the field?
What will happen if you forget to fill the field?)

File: src/main/resources/db/migration/V4__2_0_user_table.sql [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    address_id int NOT NULL,
+    description varchar(255),
+    tariff_id int NOT NULL,
+    CONSTRAINT  fk_address FOREIGN KEY (address_id) REFERENCES address(id),
+    CONSTRAINT fk_tariff FOREIGN KEY (tariff_id) REFERENCES tariff(id)
+);
+CREATE TABLE IF NOT EXISTS app_order (
+
+    id int NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name varchar(20),
```
Comment (rfisenko-integrity) [inline src/main/resources/db/migration/V4__2_0_user_table.sql]: Name of Order?)

File: src/main/java/com/yana/internship/entity/Order.java (hunk 2 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Order.java]: "order"(order by) is reserved word in SQL

Comment (rfisenko-integrity) [reply src/main/java/com/yana/internship/entity/Order.java]: Eap, exactly. Then let's use `apartment_order`. `app` looks like `application`.

File: src/main/java/com/yana/internship/entity/Address.java (hunk 1 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Address.java]: Fixed

File: src/main/java/com/yana/internship/entity/Order.java (hunk 2 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Order.java]: Done

File: src/main/java/com/yana/internship/entity/Order.java (hunk 3 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Order.java]: Changed to Date

File: src/main/java/com/yana/internship/entity/Order.java (hunk 4 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Order.java]: Changed to CascadeType.Merge

File: src/main/java/com/yana/internship/entity/Order.java (hunk 5 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Order.java]: Changed to CascadeType.Merge

File: src/main/java/com/yana/internship/entity/Tariff.java (hunk 6 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Tariff.java]: Done

File: src/main/java/com/yana/internship/entity/Tariff.java (hunk 7 above)
Comment (YanaDubov) [reply src/main/java/com/yana/internship/entity/Tariff.java]: Done

File: src/main/resources/db/migration/V1__1_0__my_first_migration.sql (hunk 8 above)
Comment (YanaDubov) [reply src/main/resources/db/migration/V1__1_0__my_first_migration.sql]: Done

File: src/main/resources/db/migration/V4__2_0_user_table.sql (hunk 10 above)
Comment (YanaDubov) [reply src/main/resources/db/migration/V4__2_0_user_table.sql]: I deleted this field

**Pull request merged**